dev/NumberBox/NumberBoxParser.cpp

```diff
@@ -0,0 +1,295 @@
+        {
+            // Read next token so that the string is parsed as a numeric token.
+            m_index++;
+            SkipWhiteSpace();
+            ss << m_inputString[m_index];
+        }
+    }
+
+    // Return token that is +,-,*,^,/
+    if (IsOperator(ss.str()))
+    {
+        m_index++;
+        m_lastToken = MathToken(MathToken::TokenType::Operator, ss.str());
+    }
+    // Return parenth token
+    else if (m_inputString[m_index] == '(')
+    {
+        m_index++;
+        m_lastToken = MathToken(MathToken::TokenType::ParenLeft, ss.str());;
+    }
+    else if (m_inputString[m_index] == ')')
+    {
+        m_index++;
+        m_lastToken = MathToken(MathToken::TokenType::ParenRight, ss.str());
+    }
+    // Numeric Tokens
+    else if (IsNumeric(std::wstring_view(ss.str())))
+    {
+        // Keep Reading until all of number is parsed
+        while (IsNumeric(std::wstring_view(ss.str())) && m_index < m_inputLength)
+        {
+            ss << m_inputString[++m_index];
+        }
+        m_lastToken = MathToken(MathToken::TokenType::Numeric, ss.str().substr(0, ss.str().size() - 1));
+    }
+    else
+    {
+        m_index++;
+        m_lastToken = MathToken(MathToken::TokenType::Invalid, ss.str());
+    }
+    return m_lastToken;
+
+}
+
+// Peek at the next token that will be returned. Used for negative checks.
+MathToken MathTokenizer::PeekNextToken()
+{
+    // Save statics that need to be restored
+    MathToken oldLastToken = m_lastToken;
+    int oldIndex = m_index++;
+    MathToken nextToken = GetToken();
+    // Restore statics
+    m_index = oldIndex;
+    m_lastToken = oldLastToken;
+    return nextToken;
+}
+
+void MathTokenizer::SkipWhiteSpace()
+{
+    while (m_inputString[m_index] == ' ' && m_index <= m_inputLength)
+    {
+        m_index++;
+    }
+}
+
+bool MathTokenizer::IsNumeric(std::wstring_view in)
+{
+    std::wregex r(L"^-?(\\d?)+(\\.)?(\\d?)+?$");
+    return (std::regex_match(in.data(), r));
+}
+
+bool MathTokenizer::IsOperator(std::wstring_view in)
+{
+    std::wregex r(L"^\\+|-|\\*|\\/|\\^$");
+    return (std::regex_match(in.data(), r));
+}
+
+// Returns 1 for op1 higher precedence, -1 for lower, -2 for error
+NumberBoxParser::OperatorPrecedence NumberBoxParser::CmpPrecedence(wchar_t op1, wchar_t op2)
+{
+    const std::wstring ops = L"-+/*^";
+    int op1prec = (int) ops.find(op1) / 2;
+    int op2prec = (int) ops.find(op2) / 2;
+    if (op1prec == std::wstring::npos || op2prec == std::wstring::npos)
+    {
+        return OperatorPrecedence::Error;
+    }
+    if (op1prec > op2prec)
+    {
+        return OperatorPrecedence::Higher;
+    }
+    else if (op2prec > op1prec)
+    {
+        return OperatorPrecedence::Lower;
+    }
+    else
+    {
+        return OperatorPrecedence::Equal;
+    }
+}
+
+// Converts an infix mathematical expression into a postfix one, whitespace separation.
```

Comment:
If we already have it in a tidy stack data structure... can we just use that input directly in the later function? I don't understand why it's converting it to a string except that [it's what I see here](https://en.wikipedia.org/wiki/Shunting-yard_algorithm).